### DISCUSSION TITLE
json: How to describe array of array of int

### DISCUSSION THREAD

**Discussion by yashi** [zephyrproject-rtos/zephyr#38711] [Q&A]
```
[[1, 2, 3],
 [10, 20, 30]]
```

How can I describe the data above in `data/json.h`? I'd like to know both conversion.
- string -> struct
- struct -> string

**Comment by d-soliveira** [comment]
Hi yashi,

Have you taken a look at [JSON_OBJ_DESCR_ARRAY_ARRAY](https://docs.zephyrproject.org/latest/reference/misc/index.html#c.JSON_OBJ_DESCR_ARRAY_ARRAY ) ?

**Reply by yashi** [reply]
Yes, I did.  The macro is actually array of array of objects as you can see from [`test_json_decoding_array_array()`](https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/lib/json/src/main.c#L223).  IOW,
````json
[
	[{
		"height": 168,
		"name": "Simón Bolívar"
	}],
	[{
		"height": 173,
		"name": "Pelé"
	}],
	[{
		"height": 195,
		"name": "Usain Bolt"
	}]
]
````
Do you have any idea how to replace the objects with integer primitive?
